pkg/netceptor/netceptor.go

```diff
@@ -730,6 +734,25 @@ func (s *Netceptor) GetServerTLSConfig(name string) (*tls.Config, error) {
 	return sc.Clone(), nil
 }
 
+// AddWorkCommand records a work command so it can be included in service announcements
+func (s *Netceptor) AddWorkCommand(command string) error {
+	if command == "" {
+		return fmt.Errorf("must provide a name")
+	}
+	s.serviceAdsLock.Lock()
+	defer s.serviceAdsLock.Unlock()
+	if n, ok := s.serviceAdsReceived[s.NodeID()]; ok {
+		// if it's the local node, just update the local service advertisement
+		// structs directly
+		for _, ad := range n {
+			ad.WorkCommands = append(ad.WorkCommands, command)
+		}
+	} else {
```

Comment:
trying to wrap my head around this

This function will only update the workcommands that are defined _locally_. Therefore I don't think the distinction between local / remote make much sense (i.e. I don't think this `else` ever hits)

If that is the case, we don't need a` workCommands [] string` on the netceptor struct -- it doesn't actually do anything anymore. We only need it on the `ServiceAdvertisement` struct since we are updating it directly via this `AddWorkCommand` function.

Reply:
Let's say we have (over TCP):

`a <------- b`

When receptor a runs this code, it'll fall into the local block and update the "local advertisement" that gets auto-created when the process starts up as part of the `listen()` call.

When receptor `b` runs this code, it'll record on the private `workCommands` field.  *This* field is what gets looked at then `b` compose the advertisement it will *send* to `a` (momentarily).